fix crash due to some fonts having family null

mariosvlad opened the pull request:
we had some issues with some fonts that had "null" family and caused the app (electron) to crash, this fixes it. I'm not experienced with Objective-C so feel free to reject this PR and do a better fix if there is a better solution.

Thanks

mariosvlad committed:
fix crash due to some fonts having family null

devongovett commented on src/FontManagerMac.mm:
```diff
@@ -78,6 +78,11 @@ static int convertWidth(float unit) {
   ResultSet *results = new ResultSet();
   
   for (id m in matches) {
+    NSString *family = (NSString *) CTFontDescriptorCopyAttribute((CTFontDescriptorRef) m, kCTFontFamilyNameAttribute);
```
maybe we could do this inside `createFontDescriptor` so it isn't repeated so much. we also need to remember to release the memory for this since this makes a copy.

mariosvlad replied:
Sorry for the late reply. I will close this and when I have time I will improve it based on your suggestions, thanks.

frangiskos committed:
fix(packages): Update nan package

nan v.2.2.x has problems with node v.10.x. Update nan to the latest version that is compatible with node 10.x

mariosvlad committed:
Merge pull request #1 from frangiskos/master

fix(packages): Update nan package
